testing/junit/src/main/java/com/linecorp/armeria/testing/junit/server/SelfSignedCertificateExtension.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit.server;
+
+import java.io.File;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.time.temporal.TemporalAccessor;
+import java.util.Date;
+
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.linecorp.armeria.testing.internal.SelfSignedCertificateRuleDelegate;
+
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+/**
+ * A {@link Extension} that provides a temporary self-signed certificate.
+ */
+public class SelfSignedCertificateExtension implements BeforeEachCallback, AfterEachCallback {
+    private final SelfSignedCertificateRuleDelegate delegate;
+
+    /**
+     * Creates a new instance.
+     */
+    public SelfSignedCertificateExtension() {
+        delegate = new SelfSignedCertificateRuleDelegate();
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @deprecated Use {@link #SelfSignedCertificateExtension(TemporalAccessor, TemporalAccessor)}.
+     *
+     * @param notBefore {@link Certificate} is not valid before this time
+     * @param notAfter {@link Certificate} is not valid after this time
+     */
+    @Deprecated
```

Comment:
Do we really need to leave deprecated constructors? This class is newly added for junit5, so it maybe okay to remove them. What do you think? @delegacy @line/cdtf

Reply:
+1 for removal